### ISSUE TITLE
charm-tools snap does not exist for arm64

### ISSUE THREAD

**Issue by dannf** [juju/charm-tools#328]
## Checklist

 - [X] Confirmed this is an issue with charm-tools, not charmstore-client
 - [X] Provide versions of tools used
 - [X] Described the feature or ways to replicate the issue

## What version am I running?

I ran the following command: `charm version` and got the following ouput:

```
N/A
```

I am using: Ubuntu 17.10 ('artful')

## Issue/Feature
I'm unable to install charm-tools on an arm64 box.

### I expect/expected the following
$ sudo snap install charm --classic
<charm-tools snap installs>

### What I got
$ sudo snap install charm --classic
error: cannot install "charm": snap not found

**Comment by patriciadomin** [comment]
I'm getting the same issue. I tried it from xenial and bionic :  

("aarch64 aarch64 aarch64 GNU/Linux, Ubuntu 16.04.5 LTS,   Ubuntu 18.04.1 LTS"): 

$ sudo snap install charm --classic
error: snap "charm" is not available on stable for this architecture (arm64) but exists on other
       architectures (amd64, armhf, i386).
$ sudo snap install charm 
error: snap "charm" is not available on stable for this architecture (arm64) but exists on other
       architectures (amd64, armhf, i386).

**Comment by johnsca** [comment]
Update on the progress for this.

The builder for the charm snap was created before arm64 was supported, and there is a soon-to-be-fixed bug that made it so that there was no way to add the new architecture to the job without deleting and recreating it.  We went ahead and did that, and sorted out the ownership at the same time, but that led us to an issue with a build error introduced with snapcraft 2.43 which caused an "out of date" error.  With the help of the snap store team, we got a workaround in place and a fix is working its way through upstream.  Unfortunately, there was then an unrelated issue with gopkg.in affecting the building of all snaps with go parts.  This has been resolved and so the build should be working at this point.

However, we are still getting failing builds with this less than helpful error:

```
Sorry, Snapcraft ran into an error when trying to running through its
lifecycle that generated a trace that has been put in '/tmp/tmpt5ncvtxx/trace.txt'.
"Submitting this error to the Snapcraft developers is not possible through the CLI
without Raven installed.
```

I'm going to continue to follow up with the snapcraft and snap store teams to get this resolved.
